commander.ts

```diff
@@ -15,6 +16,34 @@ import { Cmd, CmdVisitor, Exit, Print, Expression } from './ast/cmd';
 
 const tracer = getTracer('main');
 
+function mergeWarnings(
```

Comment:
This should probably be a method on `ParseWarning`. Maybe both `ParseWarning` and `ParseError` would have merge class methods? The reason to not make them instance methods is wanting to merge `null` warnings/errors.